compose/foundation/foundation/src/skikoTest/kotlin/androidx/compose/foundation/text/selection/CommonSelectionTest.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text.selection
+
+import androidx.compose.foundation.assertThat
+import androidx.compose.foundation.isEqualTo
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyMapping
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.key.Key
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.SkikoComposeUiTest
+import androidx.compose.ui.test.assertIsFocused
+import androidx.compose.ui.test.click
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performKeyInput
+import androidx.compose.ui.test.performMouseInput
+import androidx.compose.ui.test.pressKey
+import androidx.compose.ui.test.runSkikoComposeUiTest
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.TextFieldValue
+
+internal abstract class CommonSelectionTests(val keyboardActions: KeyboardActions, private val keyMapping: KeyMapping): KeyboardActions by keyboardActions {
+
+    abstract fun setPlatformDefaultKeyMapping(value: KeyMapping)
+
+    @ExperimentalTestApi
+    fun SkikoComposeUiTest.waitAndCheck(check: () -> Unit) {
+        waitForIdle()
+        check()
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    private fun textFieldSemanticInteraction(initialValue: String = "", semanticNodeContext: SkikoComposeUiTest.(node: SemanticsNodeInteraction, state: MutableState<TextFieldValue>) -> Unit) =
```

Comment:
The line here is too long. I believe we're supposed to fit in 100 columns.

Reply:
Do we have any config in project that I can just apply by default? I'm just using the regular formatting, assuming that if there's any, it will be applied. Don't get me wrong, I'm totally pro common code style, just don't want to waste your time and time of other reviewers to the things that can be automated.